### PR TITLE
feat(Prover CLI): `requeue` cmd

### PR DESCRIPTION
**Pull request by ilitteri**
## What ❔

Adds a `requeue` command that requeues a given batch if it's stuck. It looks at all jobs that are stuck and requeues them. A job is stuck if it has a large number of attempts and is not successful.

For now, the attempts are set via a cmd flag (`--max-attempts`) which if not set, is 10 by default.

### Usage Example

```
cd prover/prover_cli
zk f cargo run --release requeue --batch 1
```

## Why ❔

We want to be able to requeue a stuck batch  with the CLI.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.
- [x] Linkcheck has been run via `zk linkcheck`.

### CODE REVIEW

**Comment by ilitteri** [comment]
@EmilLuta we can discuss this during the daily, but I think I'm missing a `--batch` flag to avoid requeuing multiple batches jobs.

File: prover/prover_cli/src/cli.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 use clap::{command, Parser, Subcommand};
 
-use crate::commands::get_file_info;
+use crate::commands::{self, get_file_info};
```
[the remark this answers is not in the document]
Comment (ilitteri) [reply prover/prover_cli/src/cli.rs:3]: If I remove this `self` then I won't be able to use `commands::RequeueCommand` [here](https://github.com/matter-labs/zksync-era/pull/1719/files#diff-82269e3f4f5f5cb67a520a2fee051cf2f228c11201862f4721598beed1a17c91R18)

File: prover/prover_dal/src/fri_witness_generator_dal.rs [hunk 2]
```diff
@@ -295,6 +296,132 @@ impl FriWitnessGeneratorDal<'_, '_> {
...
+            RETURNING
+                l1_batch_number,
+                status
+            "#,
+        )
+        .fetch_all(self.storage.conn())
+        .await
+        .unwrap()
+        .into_iter()
+        .map(|row| WitnessJobInfo {
```
[the remark this answers is not in the document]
Comment (ilitteri) [reply prover/prover_dal/src/fri_witness_generator_dal.rs:407]: I agree, I can do the same I've done here with the `ProofCompressionJobInfo` [here](https://github.com/matter-labs/zksync-era/pull/1789/files#diff-57760fedf0f8fb551312c48f54cd4de644aa88a22b13471fde0acff94c47b7b0R250)

[the remark this answers is not in the document]
Comment (ilitteri) [reply prover/prover_dal/src/fri_witness_generator_dal.rs]: Although, this is an old method. It does not exist anymore.

**Comment by ilitteri** [comment]
I think the spellcheck error is not related to this PR:
```
error: spellcheck(Hunspell)
   --> /Users/ivanlitteri/Repositories/matter-labs/zksync-era/core/node/node_framework/src/service/runnables.rs:31
    |
 31 | ..rns a reference to created vec to satisfy the `.field` me..
    |                              ^^^
    | - Vec, vex, vecs, sec, vac, rec, vet, or one of 4 others
    |
    |   Possible spelling mistake found.

Error occurred during spell checking: Child process exited with code 1
```

File: prover/prover_dal/src/fri_witness_generator_dal.rs [hunk 3]
```diff
@@ -1395,4 +1395,106 @@ impl FriWitnessGeneratorDal<'_, '_> {
...
+                attempts: row.get::<i16, &str>("attempts") as u64,
+            })
+            .collect()
+    }
+
+    fn job_id_table_name_for(aggregation_round: AggregationRound) -> &'static str {
+        match aggregation_round {
+            AggregationRound::BasicCircuits | AggregationRound::Scheduler => "l1_batch_number",
+            AggregationRound::LeafAggregation | AggregationRound::NodeAggregation => "id",
+            AggregationRound::RecursionTip => todo!(),
```
[the remark this answers is not in the document]
Comment (ilitteri) [reply prover/prover_dal/src/fri_witness_generator_dal.rs:1497]: At that moment I didn't had the table for the recursion tip, do we have id or l1_batch_number to identify its jobs?

File: prover/prover_cli/src/commands/requeue.rs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    for stuck_job in stuck_proof_compressor_jobs {
+        println!("Re-queuing proof compressor job {stuck_job:?} 🔁",);
+    }
+
+    let stuck_prover_jobs = conn
+        .fri_prover_jobs_dal()
+        .requeue_stuck_jobs_for_batch(args.batch, args.max_attempts)
+        .await;
+
+    for stuck_job in stuck_prover_jobs {
```
[the remark this answers is not in the document]
Comment (ilitteri) [reply prover/prover_cli/src/commands/requeue.rs:80]: Done [014c88d](https://github.com/matter-labs/zksync-era/pull/1719/commits/014c88d96d0df1b0c8c4d2eb07ab02c1784cb522)

**Review by Artemka374 — APPROVED** [review]

**Pull request merged**